pages/avatar/[address].tsx

```diff
@@ -375,6 +394,178 @@ const Badge = (props: { item: Inventory.AcceptedInventoryItem }) => {
+}) => {
+  const store = useStore();
+  const modalStore = useModalStore();
+  const notificationStore = useNotificationStore();
+  const avatarSdk = useAvatarContext();
+
+  const [isClaiming, setIsClaiming] = useState(false);
+  const [fee, setFee] = useState<number>(null);
+
+  const tx = useMemo(
+    () => store.sdk.api.tx.styx.cross(),
+    [props.address, props.burnAmount],
+  );
+
+  const doClaim = async () => {
+    notificationStore.pushNotification("Minting Avatar.", {
+      type: "Info",
+      autoRemove: true,
+    });
+    notificationStore.removeNotification;
+    const response = await Avatar.claim(avatarSdk, props.address);
+    if (!response?.avatar) {
+      throw new Error((response as any).message);
+    }
+    notificationStore.pushNotification("Avatar successfully minted!", {
+      type: "Success",
+    });
+    props.onClaimSuccess();
+    modalStore.closeModal();
+  };
+
+  const onClickBurn = async () => {
+    setIsClaiming(true);
+    try {
+      if (props.hasCrossed) {
+        try {
+          await doClaim();
+        } catch (error) {
+          notificationStore.pushNotification(error.message, {
+            type: "Error",
+          });
+        }
+        setIsClaiming(false);
+      } else {
+        const signer = store.wallets.getActiveSigner() as ExtSigner;
+        await signAndSend(
+          tx,
+          signer,
+          extrinsicCallback({
+            notificationStore,
+            broadcastCallback: () => {
+              notificationStore.pushNotification("Burning ZTG.", {
+                type: "Info",
+                autoRemove: true,
+              });
+            },
+            successCallback: async () => {
+              try {
+                await delay(2000);
+                await doClaim();
+                setIsClaiming(false);
+                props.onClose?.();
+              } catch (error) {
+                notificationStore.pushNotification(error.message, {
+                  type: "Error",
+                });
+              }
+            },
+            retractedCallback: async () => {
+              setIsClaiming(false);
+            },
+            failCallback: ({ index, error }) => {
+              setIsClaiming(false);
+              notificationStore.pushNotification(
+                store.getTransactionError(index, error),
+                { type: "Error" },
+              );
+            },
+          }),
+        );
+      }
+    } catch (error) {
+      setIsClaiming(false);
+    }
+  };
+
+  useEffect(() => {
+    tx.paymentInfo(props.address).then((fee) =>
+      setFee(fee.partialFee.toJSON()),
+    );
+  }, [tx, props.address]);
+
+  useEffect(() => {
+    return () => {
+      props.onClose?.();
+    };
+  }, [props.onClose]);
+
+  return (
+    <div>
+      <div className="flex flex-1">
+        <div className="pr-6">
+          <p className="mb-4">
+            To claim your right to mint an avatar you have to pay the ferryman
```

Comment:
What's a ferryman?

Reply:
Haha its the guy who takes you across the river Styx ^^

Reply:
oh because of the pallet name? That pretty niche haha. I like the idea though, something Tarot related could be cool

Reply:
Yeah pallet is called styx with the function cross on it 🧙 😁